test/integration/targets/mas/tasks/main.yml

```
@@ -0,0 +1,151 @@
+# Test code for the mas module.
+# Copyright: (c) 2020, Lukas Bestle <project-ansible@lukasbestle.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+# Test preparation
+- name: Uninstall Rested to ensure consistent starting point
+  mas:
+    id: 421879749
+    state: absent
+  become: yes
+
+- name: Determine whether the app is installed
+  stat:
+    path: /Applications/Rested.app
+  register: install_status
+
+- name: Ensure the app is uninstalled
+  assert:
+    that:
+    - install_status.stat.exists == false
+
+- name: Wait until the OS-internal cache was updated
+  pause:
+    seconds: 5
+
+# Installation
+- name: Check if Rested needs to be installed
+  mas:
+    id: 421879749
+    state: present
+  register: install_check
+  check_mode: yes
+
+- name: Ensure that the status would have changed
+  assert:
+    that:
+    - install_check.changed == true
```

There are tests for this:
```suggestion
    - install_check is changed
```